identity/src/main/java/com/android/identity/credentialtype/CredentialType.kt

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.identity.credentialtype
+
+/**
+ * Class representing the metadata of a Credential Type
+ *
+ * @param displayName the friendly name of the Credential Type
+ * @param mdocCredentialType metadata of an mDoc Credential Type (optional)
+ * @param vcCredentialType metadata of a W3C VC Credential Type (optional)
+ *
+ */
+class CredentialType private constructor(
+    val displayName: String,
+    val mdocCredentialType: MdocCredentialType?,
+    val vcCredentialType: VcCredentialType?
+) {
+
+    /**
+     * Builder class for class [CredentialType]
+     *
+     * @param displayName the friendly name of the Credential Type
+     * @param mdocBuilder a builder for the [MdocCredentialType]
+     * @param vcBuilder a builder for the [VcCredentialType]
+     */
+    data class Builder(
+        val displayName: String,
+        var mdocBuilder: MdocCredentialType.Builder? = null,
+        var vcBuilder: VcCredentialType.Builder? = null
+    ) {
+        /**
+         * Initialize the [mdocBuilder]
+         *
+         * @param mdocDocType the DocType of the mDoc
+         */
+        fun addMdocCredentialType(mdocDocType: String) = apply {
+            mdocBuilder = MdocCredentialType.Builder(mdocDocType)
+        }
+
+        /**
+         * Initialize the [vcBuilder]
+         *
+         * @param vcType the type of the VC Credential Type
+         */
+        fun addVcCredentialType(vcType: String) = apply {
+            vcBuilder = VcCredentialType.Builder(vcType)
+        }
+
+        /**
+         * Add an attribute for both mDoc and VC, using the same identifier
+         *
+         * @param type the datatype of this attribute
+         * @param identifier the identifier of this attribute for both mDoc and VC
+         * @param displayName the friendly name of the attribute
```

Comment:
Generally, we should try and keep `displayName` to no more than 1-3 words since we want these strings to appear in consent dialogs where we have limited space. I think this is a good place to call this out...

Also want to fix up all the strings in the concrete types below... for example, for `age_in_years` we currently have *"Age attestation: How old are you (in years)?"* which probably should be *"Age in years"* instead.

Reply:
Ok, shortened a couple of display names